infrastructure/azureInfrastructure.bicep

```
@@ -0,0 +1,180 @@
+param appName string = 'vote'
+param location string = 'Norway East'
+param staticWebAppLocation string = 'westeurope'
+
+/* 
+ Using a S1 plan to support development slots.
+ Also using a S1 plan to support more than 350 concurrent web socket connections.
+ If downgrading to a cheaper plan, such as B1, you could run into issues with the number of concurrent web socket connections under peak load.
+ Also you have to remove the dev slot, and give it a separate Web App. 
+ */
+resource appServicePlan 'Microsoft.Web/serverfarms@2022-09-01' = {
+  name: '${appName}-plan'
+  location: location
+  sku: {
+    name: 'S1'
+    tier: 'Standard'
+  }
+  properties: {
+    reserved: true
+  }
+}
+
+resource voteDB 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' = {
+  name: '${appName}-db'
+  location: location
+  tags: {
+    defaultExperience: 'Azure Cosmos DB for MongoDB API'
+    'hidden-cosmos-mmspecial': ''
+  }
+  kind: 'MongoDB'
+  identity: {
+    type: 'None'
+  }
+  properties: {
+    enableFreeTier: false
+    databaseAccountOfferType: 'Standard'
+    apiProperties: {
+      serverVersion: '4.2'
+    }
+    locations: [
+      {
+        locationName: location
+        failoverPriority: 0
+        isZoneRedundant: false
+      }
+    ]
+    capabilities: [
+      {
+        name: 'EnableMongo'
+      }
+      {
+        name: 'DisableRateLimitingResponses'
+      }
+      {
+        name: 'EnableServerless'
+      }
+    ]
+    backupPolicy: {
+      type: 'Continuous'
+      continuousModeProperties: {
+        tier: 'Continuous7Days'
```

```suggestion
        tier: 'Continuous30Days'
```
Seems like this backupPolicy is required per the docs?
In that case; might 30days suffice? Let's say something bad happens at an assembly/yearly meeting for a group - how much would it help to roll back 7 days?

https://learn.microsoft.com/en-us/azure/templates/microsoft.documentdb/databaseaccounts?pivots=deployment-language-bicep#continuousmodeproperties

---

It was 7 days by default, and i guess its free? We could change it if there is something more cheap, as backup is not neccessary 

https://azure.microsoft.com/en-us/pricing/details/cosmos-db/autoscale-provisioned/

From docs:
Continuous backup is available free with a 7-day retention period, or with a 30-day retention period billed as GBs stored across all Azure regions your database is distributed to. Point-in-time restore from the database backup is billed as GBs of data restored to one Azure region. Continuous backup can be enabled on accounts using Azure Cosmos DB for NoSQL or Azure Cosmos DB for MongoDB

---

Ah of course, I thought 7 it was snapshotted every 7 days but that makes more sense after reading your docs excerpt 😅 7 days if great then